### PR TITLE
Switch to cURL for API requests in Zibal driver

### PR DESCRIPTION
**Pull request by aliqasemzadeh**
Replaced the HTTP client code with cURL to handle API requests for the Zibal payment gateway. The change aims to streamline the request process, encapsulating the data and headers setup within the cURL functions.

### CODE REVIEW

File: src/Drivers/Zibal/Zibal.php [hunk 1]
```diff
@@ -69,23 +69,32 @@ public function purchase()
...
+		}',
+            CURLOPT_HTTPHEADER => array(
+                'Content-Type: application/json'
+            ),
+        ));
+
+        $response = curl_exec($curl);
+
+        curl_close($curl);
+        $body = json_decode($response, false);
```
Comment (khanzadimahdi) [inline src/Drivers/Zibal/Zibal.php:97]: - remove `client` from class properties.
- also change `verify` to use curl.

Comment (khanzadimahdi) [reply src/Drivers/Zibal/Zibal.php:97]: any update?

Comment (aliqasemzadeh) [reply src/Drivers/Zibal/Zibal.php:97]: I will do it this week.

**Comment by aliqasemzadeh** [comment]
@khanzadimahdi 
I change all to curl.

**Comment by hadikhanian** [comment]
local.ERROR: Attempt to read property "result" on null {"exception":"[object] (ErrorException(code: 0): Attempt to read property \"result\" on null at vendor/shetabit/multipay/src/Drivers/Zibal/Zibal.php:164)

@aliqasemzadeh @khanzadimahdi

**Comment by khanzadimahdi** [comment]
@aliqasemzadeh could you please test if your changes work? I think there is a problem with your changes.

**Pull request merged**